test/modules/layers/test_quantisation.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import unittest
+
+import torch
+from torch import nn
+from torchmultimodal.modules.layers.quantisation import Quantisation
+from torchmultimodal.utils.preprocess import (
+    flatten_to_channel_vectors,
+    reshape_from_channel_vectors,
+)
+
+
+class TestQuantisation(unittest.TestCase):
+    """
+    Test the Quantisation class
+    """
+
+    def setUp(self):
+        torch.set_printoptions(precision=10)
+        torch.manual_seed(4)
+        self.num_embeddings = 4
+        self.embedding_dim = 5
+        self.encoded = torch.randn((2, self.embedding_dim, 3, 3))
```

Comment:
The same comment here about using manual calculation, e.g., how do I know we are finding the closest matching embeddings by looking at those outputs here?